### DISCUSSION TITLE
difficulty installing on Linux Mint 20.3

### DISCUSSION THREAD

**Discussion by cpufreak** [OpenLoco/OpenLoco#1392] [Q&A]
hello! I didn't want to list this as an issue as this is likely user error, but I am running Linux Mint 20.3 and I tried to follow the Ubuntu instructions, but I cannot seem to get it to work. for one, when the instructions tell me to run: 
sudo apt-get install --no-install-recommends -y \
>     pkg-config:i386 \
>     g++-multilib \
>     cmake \
>     git \
>     ninja-build \
>     libsdl2-dev:i386 \
>     libsdl2-mixer-dev:i386 \
>     libpng-dev:i386 \
>     libyaml-cpp-dev:i386
I get back the error "E: Unable to locate package libyaml-cpp-dev:i386"

after that, I did continue, and this section of the guide: 
CXXFLAGS="-m32" cmake -G Ninja \
>     -DCMAKE_BUILD_TYPE=release \
>     -DOPENLOCO_USE_CCACHE=OFF \
>     -DSDL2_DIR=/usr/lib/i386-linux-gnu/cmake/SDL2 \
>     -DSDL2_MIXER_PATH=/usr/lib/i386-linux-gnu \
>     -Dyaml-cpp_DIR=/usr/lib/i386-linux-gnu/cmake/yaml-cpp \
>     -DPNG_LIBRARY=/usr/lib/i386-linux-gnu/libpng16.so \
>     -DPNG_PNG_INCLUDE_DIR=/usr/include \
>     -DZLIB_LIBRARY=/usr/lib/i386-linux-gnu/libz.so
returns the errors: 
CMake Warning:
  No source or binary directory provided.  Both will be assumed to be the
  same as the current working directory, but note that this warning will
  become a fatal error in future CMake releases.

CMake Error: The source directory "/home/cpufreak101/openloco/OpenLoco/build" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
I still tried to go through with it, but "ninja -j$(getconf _NPROCESSORS_ONLN)" just returns "ninja: error: loading 'build.ninja': No such file or directory"

I know it's likely a lot of user error here, and I'm not *super* well versed in Linux to know my issue, so help would be greatly appreciated :D

**Comment by janisozaur** [comment]
Linux mint is an Ubuntu derivative, which removed yaml-cpp-dev:i386 since 20.04 - check if your version is based on that or later version, in which case check #1007

**Reply by cpufreak** [reply]
Appreciated, lemme give it a try and I will report results.

**Reply by cpufreak** [reply]
so first of all, Mint 20.3 is indeed based off Ubuntu 20.04.

I have looked, but it seems *way* technical for my skill level, though i did notice this thread on that link you had given me, which seems to contain some instructions.

https://github.com/OpenLoco/OpenLoco/issues/1257

I'll give it a read through, though if I'm just being dumb and doing something way wrong please let me know!

EDIT: reading through it appears I am just being dumb due to trying stuff above my skill level. I'll still give it a try, but please pardon me for any noob mistakes

EDIT 2: ok I *think* I see what I'm doing wrong, will report back.

**Reply by cpufreak** [reply]
ok, I am clearly trying something above my level. I got to the point in https://github.com/OpenLoco/OpenLoco/issues/1257 where he was getting the yaml-cpp errors, and this is where your "use code from https://github.com/OpenLoco/OpenLoco/pull/1007 " comes in, but, I literally do not know what is meant by this, I am confused, sorry to take up y'all's time.

edit: nevermind, persistence eventually led me to figure out that what he did with the "gh pr checkout 1007" required the Github cli tool, which I now have. I'm just now getting an error saying package openal is missing, I'm trying some stuff but everything I do is saying it's already installed.
